libs/data-mapper-v2/src/utils/Function.Utils.ts

```diff
@@ -258,21 +258,25 @@ export const functionDropDownItemText = (key: string, node: FunctionData, connec
   return nodeName;
 };
 
-export const getInputName = (inputConnection: InputConnection | undefined, connectionDictionary: ConnectionDictionary) => {
-  if (inputConnection) {
-    return isCustomValue(inputConnection)
-      ? inputConnection
-      : isSchemaNodeExtended(inputConnection.node)
-        ? inputConnection.node.name
-        : functionDropDownItemText(inputConnection.reactFlowKey, inputConnection.node, connectionDictionary);
+export const getInputName = (
+  inputConnection: InputConnection | undefined,
+  connectionDictionary: ConnectionDictionary
+): string | undefined => {
+  if (inputConnection && !isEmptyConnection(inputConnection)) {
+    if (isCustomValueConnection(inputConnection)) {
```

Comment:
I am seeing this logic in a multiple places (across this PR itself), can you refactor this entire thing to come from one function please? That way it will help us maintain it better